### PR TITLE
[Kubernetes] Add support for TSDB - set dimension fields for state data streams

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

Set the dimension fields for each state metric datastream. This is a follow up to the PR https://github.com/elastic/integrations/pull/5464.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Changes

The general reasons for this change can be found in https://github.com/elastic/integrations/pull/5464, as well as more information such as testing or `ecs` fields set as dimension.

Specifically, for each data stream:
- State container: `kubernetes.pod.uid` is a dimension, since it is unique across the whole cluster, as well as `kubernetes.container.id`. Every metric label is set as dimension (`phase` and `reason`) to avoid overlap on documents.
- State cronjob: `kubernetes.cronjob.name` is set as dimension, along with `kubernetes.namespace_uid` as the name is unique per namespace.
- State daemonset: for the same reason, `kubernetes.daemonset.name` and`kubernetes.namespace_uid` are dimensions.
- State deployment: `kubernetes.deployment.name` and `kubernetes.namespace_uid` are dimensions.
- State job: `kubernetes.job.name` and `kubernetes.namespace_uid` are dimensions. Note: other keyword fields inside `fields.yaml` are not set as dimension, because the `job.name` is enough to uniquely identify the document.
- State node: `kubernetes.node.name` set as dimension, since it is always present and is unique across a cluster. Note: none of the keywords inside `fields.yml` is set as dimension. This is because they were obtained through the [method label metric](https://github.com/elastic/beats/blob/ba67aa8fed619e4ef1b89c919a2dc460e531180b/metricbeat/module/kubernetes/state_node/state_node.go#L47-L53) that only stores values that are set to 1, so no overlap will happen.
- State persistent volume`: kubernetes.persistentvolume.name` set as dimension
- State persistent volume claim: `kubernetes.persistentvolumeclaim.name` set as dimension and `kubernetes.namespace_uid` as well.
- State pod: `kubernetes.pod.uid` set as dimension since it is unique. Note: none of the keywords inside `fields.yml` is set as dimension, for the same reason as stated for state node.
- State replica set: `kubernetes.namespace_uid` and `kubernetes.replicaset.name` are dimensions.
- State resource quota: All label metrics and `kubernetes.namespace` (for the unique combination name + namespace) are set as dimension.
- State service: `kubernetes.namespace_uid` and `kubernetes.service.name` are dimensions.
- State statefulset: `kubernetes.namespace_uid` and `kubernetes.statefulset.name` are dimensions.
- State storageclass: `kubernetes.storageclass.name` is set as dimension.

## Screenshots

 Before and after enabling TSDB, there was no change in the number of docs:
image

### CODE REVIEW

**Comment by MichaelKatsoulis** [comment]
> State container: kubernetes.pod.uid is a dimension, since it is unique across the whole cluster. Every metric label is set as dimension (phase and reason) to avoid overlap on documents.

@constanca-m why not the container.id? Isn't it unique across the cluster?

File: packages/kubernetes/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,14 @@
 # newer versions go on top
+- version: "1.33.1"
+  changes:
+    - description: Add support for TSDB on all state metric data streams
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/5621
+- version: "1.33.0"
```
Comment (MichaelKatsoulis) [inline packages/kubernetes/changelog.yml:7]: Here, you should  go with the 1.33.0 version. If the other commit gets merged first you will rebase/merge on this one and increment the version.

**Comment by constanca-m** [comment]
> @constanca-m why not the container.id? Isn't it unique across the cluster?

We just needed one set as dimension. I think they are both good and always available, there was no special reason to choose one over another. @MichaelKatsoulis

File: packages/kubernetes/changelog.yml (hunk 1 above)
Comment (constanca-m) [reply packages/kubernetes/changelog.yml:7]: I know, I chose it assuming the other one was being merged first. But I will adjust this if it is not case.

**Comment by MichaelKatsoulis** [comment]
> > @constanca-m why not the container.id? Isn't it unique across the cluster?
> 
> We just needed one set as dimension. I think they are both good and always available, there was no special reason to choose one over another. @MichaelKatsoulis

I asked mainly because container.id had already dimension:true and you removed it. If we go with  pod.uid then what happens if a pod has multiple containers. They will all have same pod.uid but different container.id

**Comment by constanca-m** [comment]
> I asked mainly because container.id had already dimension:true and you removed it. If we go with pod.uid then what happens if a pod has multiple containers. They will all have same pod.uid but different container.id

You're right, I hadn't remember that. I checked, and `kubernetes.container.id` is set as dimension in `fields.yaml` for state container. I haven't removed it. I will update the description to include that. @MichaelKatsoulis

**Review by gizas — APPROVED** [review]

**Review by MichaelKatsoulis — APPROVED** [review]

**Pull request merged**